### ISSUE TITLE
Trusted Publishing for MFA-level security while using CI to push

### ISSUE THREAD

**Issue by indirect** [rubygems/rubygems.org#4285]
- [x] Discuss and design the simplest possible UX for publishing gems from GitHub
- [x] Implement a streamlined web UI for gem authors publishing from GitHub, with no more than 3 required form fields, a la https://docs.pypi.org/trusted-publishers/adding-a-publisher/
- [x] Implement a GitHub action for pushing a gem that requires only one step, with zero arguments, a la https://docs.pypi.org/trusted-publishers/using-a-publisher/ https://github.com/rubygems/rubygems.org/issues/4286
- [x] Write guide documentation, walking a hypothetical user with a gem hosted on GitHub through each step (with screenshots), showing them how to publish their existing gem automatically via GitHub actions using a short-lived OIDC token
- [x] Address trusted publishing feedback from soft launch testers https://github.com/rubygems/rubygems.org/issues/4283
- [x] Write a blog post about trusted publishing and how it's great and everyone should use it https://github.com/rubygems/rubygems.github.io/issues/172
- [x] Celebrate that Trusted Publishing has publicly launched! 🎉

**Comment by ybiquitous** [comment]
Hi. Let me ask about Trusted Publishing here since I can't find where to ask.
(Please tell me a different place if my question is inappropriate 🙏🏼 )

First, my try in the suggested way by the [RubyGems document](https://guides.rubygems.org/trusted-publishing/releasing-gems/) was successful! 🎉 
(For details, see my repo's [Actions log](https://github.com/ybiquitous/easytest/actions/runs/7210080404/job/19642449449#step:4:1))

---

However, my try through a reusable workflow failed, unfortunately. I'm not sure about the failure cause, but `rubygems/configure-rubygems-credentials` failed with the following error:

```
No trusted publisher configured for this workflow found on https://rubygems.org/ for audience rubygems.org
```

See also the [failure Actions log](https://github.com/ybiquitous/easytest/actions/runs/7209639798/job/19641317427#step:4:18).

Here's a part of my Actions workflow settings:

```yaml
# .github/workflows/release.yml in the gem repository
jobs:
  release:
    uses: ybiquitous/.github/.github/workflows/ruby-release-reusable.yml@new-ruby-release
```

```yaml
# The sharable workflow in a different repository
jobs:
  release:
    runs-on: ubuntu-latest
    permissions:
      contents: write
      id-token: write
    timeout-minutes: 10
    steps:
      - name: Checkout
        uses: actions/checkout@v4
      - name: Set up Ruby ${{ inputs.ruby-version }}
        uses: ruby/setup-ruby@v1
        with:
          ruby-version: ${{ inputs.ruby-version }}
          bundler-cache: true
      - name: Publish gem to RubyGems.org
        uses: rubygems/release-gem@v1
```

For the sharable workflow code, see <https://github.com/ybiquitous/.github/pull/31/files>.

In short, **is Trusted Publishing unsupported in a sharable workflow?**

---

Last, this feature is fantastic! Thanks a lot. 👏🏼
